File: Runners/JavaRunner/java-runner/src/main/java/com/mems/manifest/TypeDescriptor.java
// TypeDescriptor.java
package com.mems.manifest;

public class TypeDescriptor {
    // kind: "primitive","array","class","nullable","task" (if needed)
    public String kind = "primitive";
    public String name; // primitive name or class FQN
    public TypeDescriptor items; // for arrays
    public TypeDescriptor of; // for nullable / task.of
}

File: Runners/KotlinRunner/src/main/kotlin/manifest/TypeDescriptor.kt
// TypeDescriptor.kt
package manifest

import com.fasterxml.jackson.annotation.JsonProperty

data class TypeDescriptor(
    // kind: "primitive","array","class","nullable","task" (if needed)
    @field:JsonProperty("Kind")
    var kind: String = "primitive",
    @field:JsonProperty("Name")
    var name: String? = null, // primitive name or class FQN
    @field:JsonProperty("Items")
    var items: TypeDescriptor? = null, // for arrays
    @field:JsonProperty("Of")
    var of: TypeDescriptor? = null // for nullable / task.of
)
